campaignion_foundation.js

```diff
@@ -156,4 +156,41 @@ Drupal.behaviors.payment_slide.attach = function (context, settings) {
   });
 };
 
+/**
+ * Provide an event to act on form step changes.
+ *
+ * If no form step is yet known, assume we are initially loading a page with a
+ * webform.
+ *
+ * You can use this event to conditionally show/hide parts of a page.
+ *
+ * NB: We currently need to use native events to allow dispatching between
+ * different versions of jQuery (`.trigger()` and `.on()` communication
+ * between the Drupal's `$` and the the mo-foundation-base's `$` does not
+ * work).
+ */
+Drupal.behaviors.form_step_data = {};
+Drupal.behaviors.form_step_data.attach = function (context, settings) {
+  // current step is always a number > 0 if the settings information is correct
+  var currentStep = settings.campaignion_foundation.webform.current_step || 1;
```

Comment:
This throws an error on pages without a webform.

Reply:
added try/catch for safety

Reply:
To me not doing any form step stuff when there is no form would make more sense?

Reply:
Returning now from behaviour, if we need to `catch` (which indicates that there is no webform on the page)

Reply:
Works now!

I’d find

```js
if (!settings.campaignion_foundation.webform) {
  return
}
var currentStep = settings.campaignion_foundation.webform.current_step || 1;
```
a bit more obvious (and it’s fewer lines of code). Are there other errors to be expected?

Reply:
`settings.campaignion_foundation` is only guaranteed to be set on webforms. so trying to access something below on pages without a webform will blow up, therefore the `try/catch` :smile:

Reply:
So then

```js
if (!(settings.campaignion_foundation && settings.campaignion_foundation.webform) {
  return
}
var currentStep = settings.campaignion_foundation.webform.current_step || 1;
```

would still be more clear, but I get why you prefer the try-catch solution…

Reply:
I think both are valid solution, I do not prefer one over the other, both are pretty clear to me. Will take your suggestion as you obviously care more about it...